sharding/types.go

```diff
@@ -23,23 +23,14 @@ package sharding
 // HashFn is a sharding hash function
 type HashFn func(identifer string) uint32
 
-// ShardScheme is a sharding scheme
-type ShardScheme interface {
-	// Shard will return a shard for a given identifer
+// ShardSet contains a sharding function and a set of shards, this interface allows for potentially out of order shard sets
+type ShardSet interface {
+	// Shard will return a shard for a given identifier
 	Shard(identifer string) uint32
 
 	// CreateSet will return a new shard set from a set of shards
 	CreateSet(shards []uint32) (ShardSet, error)
```

Comment:
Not right now, let's just add a `HashFn` however at least so you can get this back should need it for construction of other objects.  Right now not being able to get the `HashFn` out could make building utilities/whatever else an upstream change that is very easy for us to just add now as a getter.